In [1]:
#Import libraries
import numpy as np
import pandas as pd
import pickle

In [3]:
#Read data
data=pd.read_csv("/home/miano/Documents/DataGlacier/week3/flask/producer-prices_ken.csv")
data.head()

Iso3    StartDate     EndDate  Area Code           Area  \
0  #country+code  #date+start   #date+end        NaN  #country+name   
1            KEN   1991-01-01  1991-12-31      114.0          Kenya   
2            KEN   1992-01-01  1992-12-31      114.0          Kenya   
3            KEN   1993-01-01  1993-12-31      114.0          Kenya   
4            KEN   1994-01-01  1994-12-31      114.0          Kenya   

         Item Code                              Item  Element Code  \
0  #indicator+code                   #indicator+name           NaN   
1              711  Anise, badian, fennel, coriander        5539.0   
2              711  Anise, badian, fennel, coriander        5539.0   
3              711  Anise, badian, fennel, coriander        5539.0   
4              711  Anise, badian, fennel, coriander        5539.0   

                                  Element  Year Code        Year  Months Code  \
0                                     NaN        NaN  #date+year          NaN   
1  Producer Price Index (2014-2016 = 100)     1991.0        1991       7021.0   
2  Producer Price Index (2014-2016 = 100)     1992.0        1992       7021.0   
3  Producer Price Index (2014-2016 = 100)     1993.0        1993       7021.0   
4  Producer Price Index (2014-2016 = 100)     1994.0        1994       7021.0   

         Months             Unit                 Value Flag  
0           NaN  #indicator+type  #indicator+value+num  NaN  
1  Annual value              NaN             26.000000   Im  
2  Annual value              NaN             27.000000   Im  
3  Annual value              NaN             20.000000   Im  
4  Annual value              NaN             24.000000   Im

In [8]:
data=pd.read_csv(("https://raw.githubusercontent.com/hoshangk/machine_learning_model_using_flask_web_framework/master/adult.csv"),index_col=0)
data.head()

workclass  fnlwgt   education  educational-num  \
age                                                           
39           State-gov   77516   Bachelors               13   
50    Self-emp-not-inc   83311   Bachelors               13   
38             Private  215646     HS-grad                9   
53             Private  234721        11th                7   
28             Private  338409   Bachelors               13   

          marital-status          occupation    relationship    race   gender  \
age                                                                             
39         Never-married        Adm-clerical   Not-in-family   White     Male   
50    Married-civ-spouse     Exec-managerial         Husband   White     Male   
38              Divorced   Handlers-cleaners   Not-in-family   White     Male   
53    Married-civ-spouse   Handlers-cleaners         Husband   Black     Male   
28    Married-civ-spouse      Prof-specialty            Wife   Black   Female   

     capital-gain  capital-loss  hours-per-week  native-country  income  
age                                                                      
39           2174             0              40   United-States   <=50K  
50              0             0              13   United-States   <=50K  
38              0             0              40   United-States   <=50K  
53              0             0              40   United-States   <=50K  
28              0             0              40            Cuba   <=50K

In [11]:
print(data.shape)

print(data.info())
data.describe()

(32561, 14)
<class 'pandas.core.frame.DataFrame'>
Int64Index: 32561 entries, 39 to 52
Data columns (total 14 columns):
workclass          32561 non-null object
fnlwgt             32561 non-null int64
education          32561 non-null object
educational-num    32561 non-null int64
marital-status     32561 non-null object
occupation         32561 non-null object
relationship       32561 non-null object
race               32561 non-null object
gender             32561 non-null object
capital-gain       32561 non-null int64
capital-loss       32561 non-null int64
hours-per-week     32561 non-null int64
native-country     32561 non-null object
income             32561 non-null object
dtypes: int64(5), object(9)
memory usage: 3.7+ MB
None


fnlwgt  educational-num  capital-gain  capital-loss  \
count  3.256100e+04     32561.000000  32561.000000  32561.000000   
mean   1.897784e+05        10.080679   1077.648844     87.303830   
std    1.055500e+05         2.572720   7385.292085    402.960219   
min    1.228500e+04         1.000000      0.000000      0.000000   
25%    1.178270e+05         9.000000      0.000000      0.000000   
50%    1.783560e+05        10.000000      0.000000      0.000000   
75%    2.370510e+05        12.000000      0.000000      0.000000   
max    1.484705e+06        16.000000  99999.000000   4356.000000   

       hours-per-week  
count    32561.000000  
mean        40.437456  
std         12.347429  
min          1.000000  
25%         40.000000  
50%         40.000000  
75%         45.000000  
max         99.000000

The data has 14 columns, with 32561 entries. There are no null values. Mixed: Both numerical data and categorical data